src/main/java/software/amazon/encryption/s3/materials/RsaKeyring.java

```diff
@@ -146,7 +146,8 @@ private byte[] parsePseudoDataKey(DecryptionMaterials materials, byte[] pseudoDa
                 throw new S3EncryptionClientException("Invalid key length (" + dataKeyLengthBytes + ") in encrypted data key");
             }
 
-            int dataCipherNameLength = pseudoDataKey.length - dataKeyLengthBytes - 1;
+//            int dataCipherNameLength = pseudoDataKey.length - dataKeyLengthBytes - 1;
+            int dataCipherNameLength = AlgorithmSuite.ALG_AES_256_GCM_IV12_TAG16_NO_KDF.cipherName().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
nit: this is fine, as this algorithm suite is the only one supported for encryption in this library. However, strictly speaking, we shouldn't need to change the encryption code at all, since the switch from GCM to CTR only happens on decryption.

Reply:
This is part of the "authentication" conversation with Ryan